pkgs/development/libraries/qt-6/modules/qtdeclarative.nix

```diff
@@ -3,16 +3,24 @@
 , qtlanguageserver
 , qtshadertools
 , openssl
+, stdenv
 , python3
+, lib
+, pkgsBuildBuild
 }:
 
 qtModule {
   pname = "qtdeclarative";
-  propagatedBuildInputs = [ qtbase qtlanguageserver qtshadertools openssl python3 ];
+  propagatedBuildInputs = [ qtbase qtlanguageserver qtshadertools openssl ];
+  nativeBuildInputs = [ python3 ];
   patches = [
     # prevent headaches from stale qmlcache data
     ../patches/qtdeclarative-default-disable-qmlcache.patch
     # add version specific QML import path
     ../patches/qtdeclarative-qml-paths.patch
   ];
+  cmakeFlags = lib.optionals (!stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
+    "-DQt6ShaderToolsTools_DIR=${pkgsBuildBuild.qt6.qtshadertools}/lib/cmake/Qt6ShaderTools"
+    "-DQt6QmlTools_DIR=${pkgsBuildBuild.qt6.qtdeclarative}/lib/cmake/Qt6QmlTools"
```

Comment:
I think this is also correct when stdenv.buildPlatform.canExecute stdenv.hostPlatform so this could be made unconditional. Making this kind of statement unconditional helps prevent bit-rot. you can leave a comment that this is required for cross, though, so that people don't remove it thinking it's not necessary.

Reply:
Fixed. Apparently I did not need to define qt6qmltools_dir there